pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/scripts/BrokerActionProcessor.kt

```diff
@@ -0,0 +1,244 @@
+                .withSubtype(BrokerAction.Navigate::class.java, "navigate")
+                .withSubtype(BrokerAction.GetCaptchInfo::class.java, "getCaptchaInfo")
+                .withSubtype(BrokerAction.SolveCaptcha::class.java, "solveCaptcha")
+                .withSubtype(BrokerAction.EmailConfirmation::class.java, "emailConfirmation"),
+        ).add(KotlinJsonAdapterFactory())
+        .build()
+        .adapter(PirScriptRequestParams::class.java)
+    private val responseAdapter = Moshi.Builder().add(
+        PolymorphicJsonAdapterFactory.of(PirSuccessResponse::class.java, "actionType")
+            .withSubtype(NavigateResponse::class.java, "navigate")
+            .withSubtype(ExtractedResponse::class.java, "extract")
+            .withSubtype(GetCaptchaInfoResponse::class.java, "getCaptchaInfo")
+            .withSubtype(SolveCaptchaResponse::class.java, "solveCaptcha")
+            .withSubtype(ClickResponse::class.java, "click")
+            .withSubtype(ExpectationResponse::class.java, "expectation")
+            .withSubtype(FillFormResponse::class.java, "fillForm"),
+    ).add(KotlinJsonAdapterFactory())
+        .build().adapter(PirResult::class.java)
+    private var registeredActionResultListener: ActionResultListener? = null
+    private var timerJob: Job? = null
+    private var actionPushed: BrokerAction? = null
+    private var resultReceived: Boolean = false
+
+    override fun register(
+        webView: WebView,
+        actionResultListener: ActionResultListener,
+    ) {
+        actionPushed = null
+        registeredActionResultListener = actionResultListener
+        pirMessagingInterface.register(
+            webView,
+            object : JsMessageCallback() {
+                override fun process(
+                    featureName: String,
+                    method: String,
+                    id: String?,
+                    data: JSONObject?,
+                ) {
+                    processJsCallbackMessage(featureName, method, id, data)
+                }
+            },
+        )
+    }
+
+    override fun pushAction(
+        profileQuery: ProfileQuery,
+        action: BrokerAction,
+        webView: WebView,
+    ) {
+        logcat { "PIR-CSS: pushAction action: $action" }
+        resultReceived = false
+        timerJob?.cancel()
+        timerJob = coroutineScope.launch(dispatcherProvider.io()) {
+            logcat { "PIR-CSS: run timer for $action" }
+            delay(70000) // 1 minute + 10 seconds
+            logcat { "PIR-CSS: delay done" }
+            if (!resultReceived) {
+                logcat { "PIR-CSS: NO result for $action" }
+                // TODO : Might be better to retry?
+                handleError(
+                    PirErrorReponse(
+                        actionID = actionPushed?.id ?: "no-id",
+                        message = "Local timeout",
+                    ),
+                )
+                timerJob?.cancel()
```

Comment:
Do we need to explicitly cancel here, seems the end of the coroutine. It's fine if you want to leave it though